monkey/tests/unit_tests/common/event_queue/test_pypubsub_event_queue.py

```diff
@@ -0,0 +1,59 @@
+from dataclasses import dataclass
+from unittest.mock import MagicMock
+
+import pytest
+from pubsub import pub
+
+from common.event_queue.pypubsub_event_queue import PyPubSubEventQueue
+from common.events import AbstractEvent
+
+EVENT_TAG_1 = "event tag 1"
+EVENT_TAG_2 = "event tag 2"
+
+
+@dataclass(frozen=True)
+class EventType(AbstractEvent):
+    source = "1234"
+    target = None
+    timestamp = 0.0
+    tags = [EVENT_TAG_1, EVENT_TAG_2]
+
+
+@pytest.fixture(autouse=True)
+def wrap_pypubsub_functions():
+    # This is done so that we can use `.call_count` in the tests.
+    pub.sendMessage = MagicMock(side_effect=pub.sendMessage)
```

Comment:
Rather than patching `pubsub.pub`, we can just pass a `MagicMock` to `PyPubSubEventQueue`. The event queue doesn't need a real `Publisher` object.